format.py

```diff
@@ -0,0 +1,83 @@
+import re
+import sys
+from collections import defaultdict
+
+### BEFORE YOU RUN THIS SCRIPT, READ THE BELOW LINES
```

Comment:
It's good to have some notes about how to use the test.  Having a local `README.md` would be better though, to make it actual documentation.  Comments in the code are good when they really are just about the implementation.

This isn't a blocker for this PR, documentation can be added as a follow-up.